### DISCUSSION TITLE
The three cameras (INDI pylibcamera, RPI camera and RPiHQCam)  visible in Ekos-KStars Profile editor - which to choose

### DISCUSSION THREAD

**Discussion by baryah** [scriptorron/indi_pylibcamera#30] [Q&A] [answered]
I get three cameras which appear to be identical, though RPI camera and RPiHQCam are available as remote drivers and INDI pylibcamera as local driver. What is the difference between them?

I have been able to use all of them with Kstars with the RPi HQ Camera.

**Comment by scriptorron** [comment, marked as answer]
I am not sure if I understand the question right. Please correct me when I am wrong. :-)

I assume that you connected each of the 3 cameras to its own Raspberry Pi and you use a hierarchical structure of INDI servers.

The device names in EKOS seem to be hard coded for most of the driver. The old RPi Camera driver has the name "RPI Camera" and I do not know a comfortable way to change this. When you have 3 Raspberry Pi cameras in your hardware and you use "RPI Camera" for all of them you will not be able anymore to distinguish the cameras in EKOS.

That was the reason why I allowed to change the driver name of the "indi_pylibcamera" driver in the INI file (please see README.md for details). For instance I gave my guiding camera in the INI file the name "pylibcamera Guide". My main camera has the name "pylibcamera Main" in the INI file. My main camera is connected on a Raspberry Pi Zero with the name "indicam01". The server on the main camera Pi is started with
```
indiserver -v ./indi_pylibcamera.py
```
The guiding camera is connected to a Pi 3 which also runs the mount driver. The server on the Pi 3 connects to the Pi Zero with the main camera:
```
indiserver -v ./indi_pylibcamera.py indi_lx200autostar "pylibcamera Main"@indicam01.local
```
In the EKOS Profile Editor I must (!) select "Remote server" and the name of the Pi 3, independent if EKOS runs on the Pi 3 or not. In the picture below I use the IP address of the Pi 3 but you can also use its computer name or localhost if EKOS runs on the Pi 3. Selecting "Remote Server" allows free text input for the driver names. For "Camera 1" I inputted "pylibcamera Main" and for "Camera 2" "pylibcamera Guide" (use same names as in INI files here).
![grafik](https://user-images.githubusercontent.com/22291722/235306431-b890f271-1b8e-41e0-85f4-4493920d06e7.png)

That works to distinguish 2 cameras in EKOS. I do not know how you can configure a 3rd camera (maybe as "Aux 1"?).

By the way: I made some shell scripts which start the hierarchical server.
